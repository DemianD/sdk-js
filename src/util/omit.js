export default function omit(obj, keys = []) {
  // eslint-disable-next-line better/no-typeofs
  if (typeof obj !== 'object' || obj === null || Array.isArray(obj)) {
    return obj
  }

  return Object.keys(obj).reduce((result, key) => {
    if (keys.includes(key)) {
      return result
    }

    return Object.assign(result, {
      [key]: obj[key],
    })
  }, {})
}
