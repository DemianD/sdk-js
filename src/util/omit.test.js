import omit from './omit'

describe('omit', () => {
  it('should skip non objects', () => {
    expect(omit(undefined)).toBe(undefined)
    expect(omit(null)).toBe(null)
    expect(omit('')).toBe('')
    expect(omit([])).toEqual([])
  })

  it('should not omit anything if no keys to omit are given', () => {
    expect(omit({ foo: 'bar', bar: 'baz', baz: 'foo' })).toMatchSnapshot()
  })

  it('should omit the keys that are given', () => {
    expect(
      omit({ foo: 'bar', bar: 'baz', baz: 'foo' }, ['bar', 'baz'])
    ).toMatchSnapshot()
  })

  it('should return an empty object on omitting everything', () => {
    expect(
      omit({ foo: 'bar', bar: 'baz', baz: 'foo' }, ['foo', 'bar', 'baz'])
    ).toMatchSnapshot()
  })

  it('should not break when trying to omit an non-existing key', () => {
    expect(
      omit({ foo: 'bar', bar: 'baz', baz: 'foo' }, ['fake', 'bar', 'baz'])
    ).toMatchSnapshot()
  })
})
