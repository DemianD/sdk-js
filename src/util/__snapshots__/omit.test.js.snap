// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omit should not break when trying to omit an non-existing key 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`omit should not omit anything if no keys to omit are given 1`] = `
Object {
  "bar": "baz",
  "baz": "foo",
  "foo": "bar",
}
`;

exports[`omit should omit the keys that are given 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`omit should return an empty object on omitting everything 1`] = `Object {}`;
